### PR TITLE
feat: add initial CI config (elm/elixir linting/format/compile/test)

### PR DESCRIPTION
**Pull request by kyle-annen**
Initial commit for CI integration.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  Checks:
+    runs-on: ubuntu-18.04
```
Comment (7hoenix) [inline .github/workflows/ci.yml:13]: Definitely not a blocker, but curious if we are able to use ubuntu-20.04?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        with:
+          python-version: '3.9.1'
+          architecture: 'x64'
+      - uses: jorelali/setup-elm@v2
+        with:
+          elm-version: 0.19.1
+      - uses: actions/cache@v2
+        with:
+          path: deps
+          key: ${{ runner.os }}-mix-${{ hashFiles(format('{0}{1}', github.workspace, '/mix.lock')) }}
```
Comment (7hoenix) [inline .github/workflows/ci.yml:46]: 🤯  awesome!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kyle-annen) [reply .github/workflows/ci.yml:13]: Issues with Open SSL 1.1.0 and Crypto. Added card here: https://trello.com/c/v9p7Shgr/12-upgrade-ci-to-use-ubuntu-2004

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        run: mix test
+
+      - name: NPM Install
+        run: npm ci --prefix assets
+
+      - name: Gen GraphQL API
+        run: |
+          mix phx.server &
+          SERVER_PID=$! &&
+          sleep 5 &&
```
Comment (7hoenix) [inline .github/workflows/ci.yml]: I wonder if someday 5 won't be long enough 🤔 .

Comment (7hoenix) [reply .github/workflows/ci.yml]: That's a great problem to have though haha.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (7hoenix) [reply .github/workflows/ci.yml:13]: 🎉

**Comment by 7hoenix** [comment]
This is great.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (kyle-annen) [reply .github/workflows/ci.yml:46]: 🙇

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        run: |
+          npm ci --prefix assets
+          npm install -g wait-on
+
+      - name: Gen GraphQL API
+        run: |
+          mix phx.server &
+          SERVER_PID=$!
+
+          wait-on http://127.0.0.1:4000
```
Comment (kyle-annen) [inline .github/workflows/ci.yml:103]: Check it out @7hoenix !!

Comment (7hoenix) [reply .github/workflows/ci.yml:103]: Oooo :perf:

**Comment by 7hoenix** [comment]
:shipit:

**Pull request merged**